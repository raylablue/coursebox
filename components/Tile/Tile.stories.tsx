import { expect } from "@storybook/jest";
import { screen } from "@storybook/testing-library";
import { ComponentStoryObj, ComponentMeta } from "@storybook/react";
import { Tile } from "./Tile";

export default {
  title: "Content/Tile",
  component: Tile,
} as ComponentMeta<typeof Tile>;

export const BasicTile: ComponentStoryObj<typeof Tile> = {
  play: async ({ args }) => {
    await expect(screen.getByRole("heading")).toBeInTheDocument();
  },
  args: {
    header: "Lorem Ipsum",
    children:
      "Candy canes gummies cookie wafer icing sweet apple pie cookie pudding. Dragée oat cake candy canes powder toffee sugar plum pie. Muffin sweet roll jelly-o croissant candy canes.\n" +
      "Cake sweet roll biscuit gummi bears sesame snaps toffee. Tart gummies cookie pastry dessert topping brownie bonbon. Jelly beans gummi bears cookie danish candy.\n" +
      "Marzipan toffee fruitcake jelly-o chocolate cake jelly-o pie dragée wafer. Gingerbread cupcake ice cream jelly cotton candy candy canes chocolate marzipan pudding. Pie pie candy canes chocolate bar jelly sweet roll sweet roll dessert cheesecake.\n" +
      "Wafer cookie topping tootsie roll cotton candy lollipop sweet roll. Gingerbread marshmallow chupa chups cheesecake powder sesame snaps. Dessert apple pie danish jelly-o gummies cupcake icing lollipop ice cream.\n" +
      "Tootsie roll danish tiramisu croissant marshmallow muffin donut carrot cake gummies. Sugar plum gingerbread cheesecake powder shortbread muffin apple pie donut marshmallow. Fruitcake chocolate bar icing bonbon toffee chocolate bar cupcake.",
  },
};
