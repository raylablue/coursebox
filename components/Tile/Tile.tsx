import { FC } from "react";
import styled from "@emotion/styled";
import { boxShadow, borderRadius } from "../styles";

type Props = {
  /** Header string */
  header: string;
};

const Section = styled.section`
  ${borderRadius};
  padding: 1vmin 4vmin 4vmin;
  background: ${({ theme }) => theme.background};
  color: ${({ theme }) => theme.font.regular};
  ${({ theme }) =>
    boxShadow(theme.components.shadow1, theme.components.shadow2)};
`;

export const Tile: FC<Props> = ({ header, children }) => (
  <Section>
    <h2>{header}</h2>
    {children}
  </Section>
);
