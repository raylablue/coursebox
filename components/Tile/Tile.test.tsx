import React from "react";
import { render } from "@/test-utils";
import { Tile } from "@/components/Tile/Tile";

describe("Tile test cases", () => {
  it("Tile render check", () => {
    const { asFragment } = render(
      <Tile header="Lorem Ipsum">
        Lorem ipsum dolor sit amet, consecteur adipiscing elit, sed do eidsmod
        tempor incididunt ut lobore et dolore magna aliqua. Ut enim ad minim
        veniam, quis nostrud exerictation ullamco laboris.
      </Tile>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
