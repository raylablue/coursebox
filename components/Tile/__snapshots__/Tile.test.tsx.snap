// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tile test cases Tile render check 1`] = `
<DocumentFragment>
  <section
    class="css-2ex5ou"
  >
    <h2>
      Lorem Ipsum
    </h2>
    Lorem ipsum dolor sit amet, consecteur adipiscing elit, sed do eidsmod tempor incididunt ut lobore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exerictation ullamco laboris.
  </section>
</DocumentFragment>
`;
